src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -1429,7 +1429,7 @@ static void print_single_signal_handler(outputStream* st,
 // - otherwise, if this signal handler was installed by us and replaced another handler to which we
 //    are not chained (e.g. if chaining is off), print that one too.
 void PosixSignals::print_signal_handler(outputStream* st, int sig,
-                                        char* buf, size_t buflen) {
+                                        char* buf, int buflen) {
```

Comment:
This feels weird and wrong to me. Why change what are clearly memory sizes from size_t to a signed int? These are fed from both literals (OBUFLEN) and from sizeof(buf).

Reply:
Sorry, I don't like this, nor do I understand what this solves.

`size_t` is, by a deeply ingrained convention, used for memory sizes. It signals that intent clearly to the casual reader of the function declaration. `int` as size for an output buffer is away from conventions and just reads odd. Now this change perforates through all callers up to vmError and other call sites and just makes them all slightly worse, and it is pretty invasive (wrt merging changes to downports).

Reply:
I just read the other comment as well. Hats off to your pain tolerance, Coleen, I don't aim to belittle your work. I'm just really unhappy with these size_t->int changes. 

Is there really no better way to solve this? This seems to be an example where solving warnings actually makes the code base worse. Coding against conventions makes the code base brittle (because experienced developers will always use size_t and fall over this).

Reply:
Ok, I put the size_t parameter back and added the checked cast to the call to print_function_and_library_name.  Propagating size_t lower is going to require another RFE.  Too much fan-out to be part of this.

Reply:
Thank you.